### DISCUSSION TITLE
Managing/suppressing xcdat warning statements

### DISCUSSION THREAD

**Discussion by durack1** [xCDAT/xcdat#597] [2. Questions] [answered]
### Question criteria

- [X] I added a descriptive title here.
- [X] I searched the [xCDAT GitHub Discussions](https://github.com/xCDAT/xcdat/discussions) to find a similar question and didn't find it.
- [X] I searched the [xCDAT documentation](https://xcdat.readthedocs.io/en/stable/) and/or [Xarray documentation](https://docs.xarray.dev/en/stable/).

### Describe your question

We have been working on a demo notebook and are keen to suppress annoying warnings that alert users to file issues but still allow files and their contents to be opened and read.

Long story short, none of the methods that I have used in the past can suppress what originates as a [`raise ValueError`](https://github.com/xCDAT/xcdat/blob/fbf1db68fd6328d5a534e0c1e57e22e6fdd70c76/xcdat/bounds.py#L936-L939). A working example of what is happening is below

### Are there are any possible answers you came across?

I have tried several previously effective methods, the `warnings` and `contextlib` modules, but neither seems to be able to catch this and suppress it from being thrown in front of the user.

I'd be keen to understand how best to catch or suppress these `xcdat` warning statements.

### Minimal Complete Verifiable Example (MVCE)

```python
import xcdat as xc

# set inFile
inFile = "/p/user_pub/pmp/demo/sea-ice/EUMETSAT/OSI-SAF-450-a-3-0/v20231201/ice_conc_nh_ease2-250_cdr-v3p0_198801-202012.nc"

# attempt to suppress through warnings
import warnings, os
warnings.filterwarnings('ignore')
warnings.simplefilter("ignore") # Change the filter in this process
os.environ["PYTHONWARNINGS"] = "ignore" # Also affect subprocesses
fH = xc.open_dataset(inFile)

# returns
2024-02-02 04:32:11,980 [WARNING]: bounds.py(add_missing_bounds:188) >> The yc coord variable has a 'units' attribute that is not in degrees.
2024-02-02 04:32:11,980 [WARNING]: bounds.py(add_missing_bounds:188) >> The yc coord variable has a 'units' attribute that is not in degrees.

# attempt to suppress through contextlib
from contextlib import suppress
with suppress(ValueError):
    fH = xc.open_dataset(inFile)

# returns
2024-02-02 04:32:13,796 [WARNING]: bounds.py(add_missing_bounds:188) >> The yc coord variable has a 'units' attribute that is not in degrees.
2024-02-02 04:32:13,796 [WARNING]: bounds.py(add_missing_bounds:188) >> The yc coord variable has a 'units' attribute that is not in degrees.
```

### Environment

xc: 0.6.1

INSTALLED VERSIONS
------------------
commit: None
python: 3.10.10 | packaged by conda-forge | (main, Mar 24 2023, 20:08:06) [GCC 11.3.0]
python-bits: 64
OS: Linux
OS-release: 4.18.0-425.19.2.el8_7.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: en_US.UTF-8
LANG: en_US.UTF-8
LOCALE: ('en_US', 'UTF-8')
libhdf5: 1.14.0
libnetcdf: 4.9.2

xarray: 2024.1.1
pandas: 2.2.0
numpy: 1.23.5
scipy: 1.12.0
netCDF4: 1.6.3
pydap: None
h5netcdf: None
h5py: None
Nio: None
zarr: None
cftime: 1.6.3
nc_time_axis: 1.4.1
iris: None
bottleneck: None
dask: 2024.1.1
distributed: 2024.1.1
matplotlib: 3.7.1
cartopy: 0.22.0
seaborn: 0.12.2
numbagg: None
fsspec: 2023.12.2
cupy: None
pint: None
sparse: 0.15.1
flox: None
numpy_groupies: None
setuptools: 67.7.2
pip: 23.1.2
conda: None
pytest: 7.3.1
mypy: None
IPython: 8.11.0
sphinx: 5.3.0

**Comment by pochedls** [comment, marked as answer]
I think [this should work](https://stackoverflow.com/questions/11029717/how-do-i-disable-log-messages-from-the-requests-library):

```
import logging
logging.getLogger('xcdat').setLevel(logging.ERROR)
```

**Reply by durack1** [reply]
Works a treat, thanks @pochedls!
